### PR TITLE
Added random team button

### PR DESCRIPTION
**Pull request by cjdenio**
## Description

This PR adds an "I'm Feeling Lucky" button to the team list and 404 page. This button brings the user to a random team.

## Motivation and Context

Bored FRC enthusiasts and 404 page viewers will be able to indulge themselves by looking at a random team's page.

## How Has This Been Tested?

The magic unicorns tested the code, and found it free of errors. :rainbow: 
Just kidding! This feature's pretty small, and it works perfectly in the dev container.

## Screenshots (if appropriate):
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

File: controllers/team_controller.py [hunk 1]
```diff
@@ -11,6 +11,21 @@
 from renderers.team_renderer import TeamRenderer
 from template_engine import jinja2_engine
 
+import webapp2
+import random
+
+
+class RandomTeam(webapp2.RequestHandler):
+    """
+    Lovingly written by Caleb Denio (https://github.com/cjdenio)
```
Comment (dracco1993) [inline controllers/team_controller.py]: Generally, I wouldn't put this kind of thing in open source code.  Having attribution is fine, in fact it's a good idea, but that should come from a "contributors" file/page or the like, not from the individual source files.  🙂  If someone actually looking at the code needs to know who wrote it, `git blame` should suffice.

File: templates_jinja2/404.html [hunk 2]
```diff
@@ -15,6 +15,8 @@ <h2>Error 404</h2>
             <form action="/search">
                 <input type="text" class="form-control typeahead" placeholder="Search for a team, event, or match" name="q"/>
             </form>
+            <p style="margin-top:20px;margin-bottom:5px">Feeling lucky?</p>
+            <a class="btn btn-default" href="/teams/random"><span class="glyphicon glyphicon-random"></span> Show Me A Random Team</a>
```
Comment (dracco1993) [inline templates_jinja2/404.html]: Nice use of a bad situation.  🙂

File: templates_jinja2/team_list.html [hunk 3]
```diff
@@ -22,6 +22,7 @@ <h1>Teams {{cur_page_label}} <small>{{ num_teams }} teams</small></h1>
       <div class="btn-group pull-right">
         <a class="btn btn-default" href="/avatars/2020"><span class="glyphicon glyphicon-user"></span> View  Avatars</>
         <a class="btn btn-default" href="/nearby?search_type=teams"><span class="glyphicon glyphicon-search"></span> Search Nearby</a>
+        <a class="btn btn-default" href="/teams/random"><span class="glyphicon glyphicon-random"></span> I'm Feeling Lucky</a>
```
Comment (dracco1993) [inline templates_jinja2/team_list.html]: Personally, I'm not sure if this is the right place to put this, although admittingly I'm not sure where would be, other than just on the 404 page.  🤔

File: controllers/team_controller.py [hunk 4]
```diff
@@ -11,6 +11,21 @@
...
+    """
+    def get(self):
+        teams = Team.query().fetch(keys_only=True)
```
Comment (fangeugene) [inline controllers/team_controller.py]: We should probably put this behind memcache for speed and $ reasons.

File: templates_jinja2/team_list.html (hunk 3 above)
Comment (cjdenio) [reply templates_jinja2/team_list.html]: It felt right to me, but I'm open to other ideas.

File: controllers/team_controller.py (hunk 1 above)
Comment (cjdenio) [reply controllers/team_controller.py]: I'll remove it; it's no problem :smile:

File: controllers/team_controller.py (hunk 4 above)
Comment (cjdenio) [reply controllers/team_controller.py]: I'm open to changes, though I'm not sure how a random response can be cached.

File: templates_jinja2/404.html (hunk 2 above)
Comment (cjdenio) [reply templates_jinja2/404.html]: Thanks :smile:

File: controllers/team_controller.py (hunk 4 above)
Comment (lethosor) [reply controllers/team_controller.py:20]: Pretty sure the "all teams" query is what you'd want to cache.

Comment (cjdenio) [reply controllers/team_controller.py:20]: I'm not sure what our caching capacity is, but caching *every single team* sounds like a bit much. Could this leech off of the cache of the [team list page](https://www.thebluealliance.com/teams)?

**Pull request closed without merging**